Add support for on-demand subtitles

JonnyWong16 opened the pull request:
## Description

Changes:
* New methods `Video.searchSubtitles` to search for on-demand subtitles and `Video.downloadSubtitltes` to download on-demand subtitles.
* Cleaned up `uploadSubtitle` and `removeSubtitles` methods and updated doc strings.

Breaking change:
* Added `subtitleStream` as first argument to `Video.removeSubtitles`.


## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have added tests when applicable

blacktwin commented:
Flipping through the `hearingImpaired` and `forced` value combinations while searching for subtitles and `SubtitleStream()` should probably get additional attributes for `hearingImpaired` and `perfectMatch`. Both seem to not exist or exist with the value of `1`.

JonnyWong16 commented:
Added the `hearingImpaired` attribute (`bool`) which will always be `True` / `False`. This matches the `forced` attribute.

Added the `perfectMatch` attribute (`bool`) which will be `None` / `True`. I did not want this flag to be `False` because that can be misleading for subtitles that are not downloaded on-demand.
